OptimoveSDK/Sources/Classes/Optimove.swift

```diff
@@ -367,6 +367,20 @@ extension Optimove {
         Optimobile.trackIBeaconProximity(beacon: beacon)
     }
     
+    
+    /// Records a notification open event
+    /// - Parameter notificationId: the notification ID
+    @objc public func trackOpenMetric(notificationId: Int) {
+        Optimobile.trackEvent(
+            eventType: OptimobileEvent.MESSAGE_OPENED,
+            properties: [
+                "type" : KS_MESSAGE_TYPE_PUSH,
+                "id": notificationId
+            ],
+            immediateFlush: true
+        )
```

Comment:
Rather than duplicating logic here, would it be preferable to call into the existing open tracking method: https://github.com/optimove-tech/Optimove-SDK-iOS/blob/259ed73cb8b1816dce85f4baee8b435418f258c1/OptimoveSDK/Sources/Classes/Optimobile/Optimobile%2BPush.swift#L212

I think if we change the contract to take in the `aps` payload & construct a model here, it's more flexible for the future and removes the need for the consumer to learn how to parse our ID out of the payload.

WDYT? :)

Reply:
Sure, `userInfo` is a typical data structure for notifications, and it might be useful here.